orttraining/orttraining/core/optimizer/compute_optimizer/padding_elimination.cc

```diff
@@ -0,0 +1,570 @@
+        candidate_outputs.insert(cur);
+      } else {
+        subgraph.insert(cur->MutableOutputDefs()[0]);
+        PushAllOutputNode(graph, to_visit, cur, visited);
+      }
+    } else if (graph_utils::IsSupportedOptypeVersionAndDomain(*cur, "Dropout", {12, 13})) {
+      ORT_ENFORCE(subgraph.find(cur->MutableInputDefs()[0]) != subgraph.end());
+      subgraph.insert(cur->MutableOutputDefs()[0]);
+      subgraph.insert(cur->MutableOutputDefs()[1]);
+      PushAllOutputNode(graph, to_visit, cur, visited);
+    } else if (graph_utils::IsSupportedOptypeVersionAndDomain(*cur, "Cast", {9, 13})) {
+      ORT_ENFORCE(subgraph.find(cur->MutableInputDefs()[0]) != subgraph.end());
+      subgraph.insert(cur->MutableOutputDefs()[0]);
+      PushAllOutputNode(graph, to_visit, cur, visited);
+    } else if (graph_utils::IsSupportedOptypeVersionAndDomain(*cur, "MatMul", {1, 9, 13})) {
+      if (subgraph.find(cur->MutableInputDefs()[0]) != subgraph.end()) {
+        // If shape of [batch_size, seqlen, ...] is propagated from the first argument of MatMul.
+        // The dim size of the first argument must larger than 2 to propagete the first two dims to the output.
+        // Or else the first two dims of the output will not be [batch_size, seqlen] and this MatMul will be added
+        // to candidate_outputs as the output of the subgraph.
+        if (cur->InputDefs()[0]->Shape()->dim_size() > 2) {
+          subgraph.insert(cur->MutableOutputDefs()[0]);
+          PushAllOutputNode(graph, to_visit, cur, visited);
+        } else {
+          LOG_DEBUG_INFO(logger,
+                         "PaddingElimination::dim size of left input of matmul smaller than 3 and \
+                            this matmul would be output of subgraph.");
+          candidate_outputs.insert(cur);
+          continue;
+        }
+      } else if (subgraph.find(cur->MutableInputDefs()[1]) != subgraph.end()) {
+        LOG_DEBUG_INFO(logger, "PaddingElimination::right edge of matmul would not included.");
+        candidate_outputs.insert(cur);
+        continue;
+      } else {
+        ORT_THROW("");
+      }
+    } else if (graph_utils::IsSupportedOptypeVersionAndDomain(*cur, "PythonOp", {1}, kMSDomain)) {
+      if (subgraph.find(cur->MutableInputDefs()[0]) == subgraph.end()) {
+        candidate_outputs.insert(cur);
+        continue;
+      }
+      auto func_name = static_cast<std::string>(cur->GetAttributes().at("name").s());
+      if (func_name == "_InspectActivation" || func_name == "_IncrementStep") {
+        subgraph.insert(cur->MutableOutputDefs()[1]);
+        PushAllOutputNode(graph, to_visit, cur, visited);
+      } else {
+        candidate_outputs.insert(cur);
+      }
+    } else {
+      candidate_outputs.insert(cur);
+    }
+  }
+}
+}  // namespace
+
+Status PaddingElimination::ApplyImpl(Graph& graph, bool& modified, int graph_level, const logging::Logger& logger) const {
+  LOG_DEBUG_INFO(logger, "Enter PaddingElimination");
+
```

Comment:
add an INFO log similar to this, for better checking how many nodes affected by the graph transformer. 

```
LOGS(logger, INFO) << "Total handled SCE node count:  " << handled_sce_node_count;
```

Reply:
done